retail-managment-system/rms-ui/src/app/pages/transaction/transaction.component.ts

```diff
@@ -0,0 +1,148 @@
+import { Component, OnInit } from '@angular/core';
+import {TransactionRepository} from 'src/app/domain/transaction/transaction.repository'
+import { Transaction, transactionType } from 'src/app/domain/transaction/models/transaction'
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { Investors } from '../../domain/investors/models/investor'
+import { InvestorsRepository } from 'src/app/domain/investors/investor.repository';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { MatDialog } from '@angular/material/dialog';
+import { ConfirmDialogComponent } from '../../shared/components/confirm-dialog/confirm-dialog.component';
+
+ 
+
+const ELEMENT_DATA : Transaction[] = []
+
+@Component({
+  selector: 'app-transaction',
+  templateUrl: './transaction.component.html',
+  styleUrls: ['./transaction.component.scss']
+})
+export class TransactionComponent implements OnInit {
+
+  transactionPostForm : FormGroup
+  allTransactions : Transaction[] = []
+  investorNames : Investors[]=[]
+  transactionTypeEnum = Object.values(transactionType)
+  displayedColumns: string[] = ['id','investorName','type','amount','date','edit-icon' ,'delete-icon'];
+  dataSource = ELEMENT_DATA
+  currentData!: Transaction
+  isButtonVisible: boolean = true
+  submit: boolean = false;
+  index =0
+
+
+  constructor(
+    private investorsRepository : InvestorsRepository,
+    private transactionRepository : TransactionRepository,
+    private fb :FormBuilder,
+    private SnackBar: MatSnackBar,
+    private dialog: MatDialog
+    ) {
+    this.transactionPostForm = this.fb.group({
```

Comment:
write this(form) in function to achieve single responsibility principle and call this in ngOnInit()

Reply:
done